pkg/privilege/privileges/ldap/ldap_common.go

```diff
@@ -191,13 +220,19 @@ func (impl *ldapAuthImpl) getConnection() (*ldap.Conn, error) {
 			Password: impl.bindRootPWD,
 		})
 		if err != nil {
+			logutil.BgLogger().Warn("fail to use LDAP connection bind to anonymous user. Retrying", zap.Error(err),
+				zap.Duration("backoff", getConnectionRetryInterval))
+
 			// fail to bind to anonymous user, just release this connection and try to get a new one
 			impl.ldapConnectionPool.Put(nil)
 
 			retryCount++
 			if retryCount >= getConnectionMaxRetry {
 				return nil, errors.Wrap(err, "fail to bind to anonymous user")
 			}
+			// Be careful that it's still holding the lock of the system variables, so it's not good to sleep here.
+			// TODO: refactor the `RWLock` to avoid the problem of holding the lock.
```

Comment:
You can use ```syncutil.RWMutex``` to replace ```sync.RWMutex```. which can find deadlock in the test.

Reply:
The current issue is not about deadlock (there's only one lock, and is not recursive, so it's impossible to deadlock here).

The problem is that a pending write lock (by rebuildSysVarCache) will block all other read lock, which I didn't realize before 🤦 😢 .